### PR TITLE
Eip4788 : SystemTx Refactor

### PR DESCRIPTION
**Pull request by Demuirgos**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Changes

- _List the changes_

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by MarekM25 — COMMENTED** [review]
I think we should target master. WDYT?

**Comment by Demuirgos** [comment]
> I think we should target master. WDYT?

yeah, I was thinking about to avoid the double hoops maintaining two branches and their dependencies

File: src/Nethermind/Nethermind.Evm/TransactionProcessing/SystemTxProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+
+            if (!ValidateStatic(tx, header, spec, tracer, opts, out long intrinsicGas))
+                return;
+
+            UInt256 effectiveGasPrice =
+                tx.CalculateEffectiveGasPrice(spec.IsEip1559Enabled, header.BaseFeePerGas);
+
+            if (opts == ExecutionOptions.Commit || opts == ExecutionOptions.None)
+            {
+                decimal gasPrice = (decimal)effectiveGasPrice / 1_000_000_000m;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Evm/TransactionProcessing/SystemTxProcessor.cs:90]: I guess you could reuse more code. Let's discuss it.

Comment (Demuirgos) [reply src/Nethermind/Nethermind.Evm/TransactionProcessing/SystemTxProcessor.cs:90]: Yes, I am planning on reusing as much as possible from code, just trying to find a way that doesn't make the execution life cycle abit convoluted

**Pull request closed without merging**